### PR TITLE
P0009r8: Fix various issues including #73

### PR DESCRIPTION
**Pull request by mhoemmen**
@crtrott @dsunder @dhollman

### CODE REVIEW

File: P0009/P0009.bs [hunk 1]
```diff
@@ -1753,16 +1853,22 @@ a contiguous set of objects of type *ElementType* are accessed.
...
-and offset operations to an accessor.
-The subscript operation returns an object which provides access to the indexed element,
-this return type may not be `T&`.
-The offset operation returns an accessor for the contiguous subset of elements beginning at the offste value,
-this return type may not be the same accessor type..
-An accessor constructor may impose additional restrictions on the 
-contiguous set of objects; e.g., restrict `T` to be trivially copyable or have a wider alignment.
+An accessor policy defines
+a type through which a single element is accessed,
+a type through which the contiguous set of elements is accessed,
```
Comment (mhoemmen) [inline P0009/P0009.bs]: What does "through" mean?

Comment (mhoemmen) [reply P0009/P0009.bs]: "defines a generalization of pointer, which is a handle to the contiguous set of elements."  (relax contiguity later, in a later paper, e.g., for segmented allocations)

Comment (mhoemmen) [reply P0009/P0009.bs]: "defines a reference, which is a handle to a single element"

File: P0009/P0009.bs [hunk 2]
```diff
@@ -1797,7 +1903,7 @@ Table �: Accessor policy requirements
 <tr>
   <td>`A::offset_policy`</td>
   <td></td>
-  <td>Accessor policy which can access a pointer that has been return by `a.offset(p,i)`.</td>
+  <td>Accessor policy which can access a pointer that has been returned by `a.offset(p,i)`.</td>
 </tr>
 <tr>
   <td>`a.decay(p)`</td>
```
Comment (mhoemmen) [inline P0009/P0009.bs]: this is another relaxation point, only required under certain circumstances; returns a span

File: P0009/P0009.bs [hunk 3]
```diff
@@ -1797,7 +1903,7 @@ Table �: Accessor policy requirements
 <tr>
   <td>`A::offset_policy`</td>
   <td></td>
-  <td>Accessor policy which can access a pointer that has been return by `a.offset(p,i)`.</td>
+  <td>Accessor policy which can access a pointer that has been returned by `a.offset(p,i)`.</td>
```
Comment (mhoemmen) [inline P0009/P0009.bs]: "Accessor policy whose pointer typedef is convertible from `a.offset(p,i)`"

**Comment by mhoemmen** [comment]
Per request by @crtrott, I will make another pass that turns most of the code in `mdspan.subspan` into math, and submit that as a separate pull request (but with all the other changes in this branch).

File: P0009/P0009.bs [hunk 4]
```diff
@@ -2398,59 +2506,214 @@ subspan
...
+for 0 <= `k` < &rho;, r does not equal `get_seq<k>(rank_map)`.
+
+Let `sub` be the return value of `subspan(src, slices...)`.
+
+* *Requires:*
+    * `sizeof(slices...)` equals `src.rank()`.
+    * For all `r` in 0, 1, ..., `sizeof...(slices)-1`,
+      `0 <= first[r] && first[r] < last[r] && last[r] <= src.extent(r)` is `true`.
+* *Constraints:*
+    * For all types `T` in the parameter pack `SliceSpecifiers...`,
```
[the remark this answers is not in the document]
Comment (mhoemmen) [reply P0009/P0009.bs]: Fixed in P0009r8-more-math branch.

File: P0009/P0009.bs [hunk 5]
```diff
@@ -2398,59 +2506,214 @@ subspan
...
-    * If `src.is_strided()` then `sub.is_strided()` and `sub.stride(k)==src.stride(R[k])` <br/>
-    * If `src.static_extent(R[k]) != dynamic_extent` and `slices[R[k]]` is convertable to `all_type` then `sub.static_extent(k) == src.static_extent(R[k])`.
-<br/>
+Let &rho; equal `sub.rank()`.
 
-* Examples:
+Denote the value of the <math>k</math>-th member of `slices...`
+by <math>s<sub>k</sub></math>.
+
+Define `rank_map` as the length &rho; `std::integer_sequence` of `ptrdiff_t` consisting of all k such that `is_convertible_v<`<math>s<sub>k</sub></math>`, pair<ptrdiff_t, ptrdiff_t>> || is_convertible_v<`<math>s<sub>k</sub></math>`, all_type>` is `true`.
```
[the remark this answers is not in the document]
Comment (mhoemmen) [reply P0009/P0009.bs]: Fixed in P0009r8-more-math branch.

**Comment by mhoemmen** [comment]
New PR is #77.  I am in favor of closing this PR (#74) and focusing on #77.

**Comment by dhollman** [comment]
Agreed. Closing to avoid confusion.

**Pull request closed without merging**